### ISSUE TITLE
Calling notifyListeners(x) in succession has a weird behaviour

### ISSUE THREAD

**Issue by egorbelibov** [VGVentures/property_change_notifier#8]
In short, when calling `notifyListeners` in succession, seems like only the last one triggers an update:

``` 
notifyListeners(x);
notifyListeners(y);
notifyListeners(z); // Only the widget subscribed to z is notified
```

I've created a plain Flutter project that conveys the issue: 

#### Steps to reproduce

1. Download [Compressed Project](https://drive.google.com/file/d/1l7qLGMpp8nVHD54WA8pmgUnU63yrjAVL/view?usp=sharing) ([View App Screenshot](https://drive.google.com/file/d/1M7hW-04KcAQxlICv3sdUnH_29h-llaje/view))
2. (In your CLI) Go to folder
3. (In your CLI) Run: `flutter create .` 
4. (In your CLI or Editor) Run project: `flutter run` 
5. Test using the two buttons ('**Update One Widget**' and '**Update Multiple Widgets**'). As you'll see, both of them have the same effect.

#### Project rundown

The project is very simple. It has grid of 15 `Box` widgets and two buttons:
- **Update One Widget** - Generates a random id (0..14) and a random color and passes them to `AppModel`. 
- **Update Multiple Widgets** - Generates multiple randomIds (0..14) and multiple random colors and passes them to `AppModel`. 

The project uses a `PropertyChangeProvider` to provides a simple `AppModel` through the entire `App`.

`AppModel` only has a `List<BoxModel>` initialized in the constructor and two methods that update the list and notify changes:  

``` 
// Called by the singleUpdateButton() in app.dart
//
// WORKING AS EXPECTED
void updateBox({@required id, @required Color color}) {
  boxes[id] = BoxModel(id: id, color: color);
  notifyListeners(id);
}

// Called by the multipleUpdatesButton() in app.dart
//
// NOT WORKING AS EXPECTED
void updateBoxes(List<BoxModel> newBoxValues) {
  newBoxValues.forEach((box) {
    boxes[box.id] = BoxModel(id: box.id, color: box.color);
    notifyListeners(box.id);
  });
}
```

`App` is a **StatefullWidget** that overrides `onInitState()` and calls `initializeValues()` to get access to the boxes info and render each `Box` using a grid.

``` 
void initializeValues() {
  appModel = PropertyChangeProvider.of<AppModel>(context, listen: false).value;
  boxes = appModel.boxes;
}

List<Widget> renderBoxes() {
  return boxes.map((box) => Box(box.id)).toList();
}
```

Finally, each `Box` is a **StatefullWidget** and is subscribed to changes using `PropertyChangeProvider.of<AppModel>` with its `id` as a property.
``` 
class Box extends StatefulWidget {
  final int id;
  Box(this.id);

  @override
  _BoxState createState() => _BoxState();
}

class _BoxState extends State<Box> {
  BoxModel box;

  @override
  Widget build(BuildContext context) {
    box = PropertyChangeProvider.of<AppModel>(
      context,
      listen: true,
      properties: [widget.id],
    ).value.boxes[widget.id];

    print('${widget.id} - ${DateTime.now()}');

    return Padding(
      padding: const EdgeInsets.all(4.0),
      child: Container(
        color: box.color,
      ),
    );
  }
}
```

**Comment by martinrybak** [comment]
Thanks, I've identified the issue and am working on a fix.

**Comment by egorbelibov** [comment]
Thanks @martinrybak. When do you reckon you'll be able to publish a fixed version to **pub.dev**?

**Comment by martinrybak** [comment]
Fixed in version `0.2.0`

**Comment by martinrybak** [comment]
FYI this required a small breaking API change. The result of `PropertyChangeProvider.of()` now has a set of `properties` instead of a single `property` value.
